### ISSUE TITLE
incorrect docs for how to disable scala incremental compilation

### ISSUE THREAD

**Issue by dsilvasc** [gradle/gradle#12631]
### Expected Behavior

Setting `scalaCompileOptions.isForce = true` should disable incremental compilation per docs and javadoc:

https://docs.gradle.org/6.3/userguide/scala_plugin.html#sec:scala_incremental_compilation
https://docs.gradle.org/6.3/javadoc/org/gradle/language/scala/tasks/BaseScalaCompileOptions.html#isForce--

### Current Behavior

It doesn't.  Gradle checks the analysis file instead of that property. See isNonIncrementalCompilation in AbstractScalaCompile as of Gradle 6.3:
https://github.com/gradle/gradle/blob/b8d50a434b485d6111aab1635ba8e271caf2c3a5/subprojects/language-scala/src/main/java/org/gradle/language/scala/tasks/AbstractScalaCompile.java#L109

### Context

Couldn't disable incremental compilation for the compileScoverageScala task from https://github.com/scoverage/gradle-scoverage

### Steps to Reproduce 

```kotlin
        tasks.named<ScalaCompile>(the<JavaPluginConvention>().sourceSets.getByName("scoverage").getCompileTaskName("scala")).configure {
            scalaCompileOptions.isForce = true
        }
```

Then run `./gradlew compileScoverageScala -i` for a (sub-)project with at least 2 files, where one of them uses definitions from the other.

You'll get some output like this, followed by a compile error from one of those definitions not being found:

```
  Compiling with Zinc Scala compiler.
  Prepared Zinc Scala inputs: 0.055 secs
  Compiling 1 Scala source to your-sub-project/build/classes/scala/scoverage ...
```

### Your Environment
Gradle 6.3.

**Comment by rpalcolea** [comment]
@ljacomet @dsilvasc 

My understanding is that `CleaningJavaCompiler`, is responsible for removing stale classes

Still, this piece in the zinc scala compiler should take care of the force https://github.com/gradle/gradle/blob/45b67bc956a921b8691882af1d4662ec2ccb248c/subprojects/language-scala/src/main/java/org/gradle/api/internal/tasks/scala/ZincScalaCompiler.java#L140 regardless if CleaningJavaCompiler is used or not 


The compiler spec has the scala options https://github.com/gradle/gradle/blob/b8d50a434b485d6111aab1635ba8e271caf2c3a5/subprojects/language-scala/src/main/java/org/gradle/language/scala/tasks/AbstractScalaCompile.java#L128 and it’s created here https://github.com/gradle/gradle/blob/1bee8f57a68cfffc42125b97e5658bba12558e45/subprojects/scala/src/main/java/org/gradle/api/tasks/scala/ScalaCompile.java#L134

The scala compiler is created in the factory https://github.com/gradle/gradle/blob/45b67bc956a921b8691882af1d4662ec2ccb248c/subprojects/scala/src/main/java/org/gradle/api/internal/tasks/scala/ScalaCompilerFactory.java#L61

I noticed that `ZincScalaCompilerFacade` is used for the `DaemonScalaCompiler` as the compiler
Class https://github.com/gradle/gradle/blob/0893da4d230166803874dd8326830625eabeb7ce/subprojects/language-scala/src/main/java/org/gradle/api/internal/tasks/scala/DaemonScalaCompiler.java#L68

This facade is responsible for creating the `ZincScalaCompiler` https://github.com/gradle/gradle/blob/45b67bc956a921b8691882af1d4662ec2ccb248c/subprojects/language-scala/src/main/java/org/gradle/api/internal/tasks/scala/ZincScalaCompilerFacade.java#L47. 

I’d think things are just fine with docs in this case and behavior should be the expected when using `isForce`

**Comment by ljacomet** [comment]
The documentation is actually correct, the implementation is not at all. When Gradle 6 migrated to SBT Zinc, a bug was introduced around the handling of `force` (as in force recompilation).

This should be fixed through #13224 and #13238

**Comment by ljacomet** [comment]
Closing as the fix has been merged.
